src/lightning_app/runners/cloud.py

```diff
@@ -329,6 +332,20 @@ def dispatch(
                         default=True,
                     )
                     app_config.cluster_id = None
+                if existing_instance and existing_instance.spec.cluster_id != app_config.cluster_id:
```

Comment:
I am not sure to fully understand why it is needed.

Reply:
It is a single running instance per cluster, not over all clusters.

Reply:
cc @edenlightning

Reply:
It's my understanding that we want to have only one instance of the app running per project.

It's not a problem when updating apps on a single cluster since we stop any previous instance before launching a new one.

When launching an app instance onto a new cluster, this isn't the case. If we launched multiple instances of the app, it could lead to issues with `lightning connect app`, and existing data may not be available to the new cluster which may affect the app.

So instead of having multiple app instances per project / cluster, we decided to throw an error. (@lantiga and @edenlightning please correct me if I'm wrong. )

Reply:
So yes, we 100% need to make the app name unique across a project, independently on whether it is running on cluster X or cluster Y.

We decided to throw an error because making a mistake with clusters may mean that an app meant to be on a VPC is now running on a public cluster. This is safer but it's kind of suboptimal from a UX standpoint IMO, because an app is an app no matter what cluster it is running on, and our journey shouldn't depend on where it is running.

But I think that's ok for now and we can change it later if we/users get annoyed or confused.